webapp/sources/rudder/rudder-web/src/main/scala/com/normation/rudder/web/snippet/HealthcheckInfo.scala

```diff
@@ -0,0 +1,133 @@
+* Related Module is not considered as a part of the work and may be
+* distributed under the license agreement of your choice.
+* A "Related Module" means a set of sources files including their
+* documentation that, without modification of the Source Code, enables
+* supplementary functions or services in addition to those offered by
+* the Software.
+*
+* Rudder is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Rudder.  If not, see <http://www.gnu.org/licenses/>.
+*
+*************************************************************************************
+*/
+
+package com.normation.rudder.web.snippet
+
+import bootstrap.liftweb.RudderConfig
+import com.normation.rudder.services.healthcheck.HealthcheckResult
+import com.normation.rudder.services.healthcheck.HealthcheckResult.Critical
+import com.normation.rudder.services.healthcheck.HealthcheckResult.Ok
+import com.normation.rudder.services.healthcheck.HealthcheckResult.Warning
+import com.normation.rudder.services.healthcheck.HealthcheckUtils.compareCheck
+import com.normation.zio._
+import net.liftweb.common.Loggable
+import net.liftweb.http.DispatchSnippet
+
+import scala.xml.NodeSeq
+
+class HealthcheckInfo(
+) extends DispatchSnippet with Loggable{
+
+  def dispatch = {
+    case "render"     => healthcheckInfo
+    case "renderIcon" => healthcheckInfoIcon
+  }
+
+  def isCriticalOrWarning(checks: List[HealthcheckResult]): (Boolean, Boolean) = {
+    val isCritical: Boolean = checks.exists(
+      x => x match {
+        case Critical(_) => true
+        case _           => false
+      }
+    )
+      val isWarning: Boolean = checks.exists(
+        x => x match {
+          case Warning(_) => true
+          case _          => false
+        }
+      )
+    (isCritical, isWarning)
+  }
+
+  def healthcheckInfo(html: NodeSeq) ={
+    def notifHtml (notifClass:String , notifTitle:String, checks: List[HealthcheckResult]): NodeSeq ={
+      val liChecks = checks.map(c => {
+        val circle = c match {
+          case _:Warning  => "<div class='circle-notif warning-light-notif'></span>"
+          case _:Critical => "<div class='circle-notif critical-light-notif'></span>"
+          case _          => ""
+        }
+        "<li class='notif-msg-hc'><span>" + c.msg + circle + "</span></li>"
+      })
+      val checksMenu = "<ul class='menu'>" + liChecks.mkString("\n") + "</ul>"
+
+      val tooltipContent = "<h4><i class='fa fa-warning'></i> "+ notifTitle +"</h4><div>" + checksMenu + "</div>"
+      <li
+      class={"plugin-warning " + notifClass}
+      data-toggle="tooltip"
+      data-placement="bottom"
+      data-html="true"
+      data-original-title={tooltipContent}
```

Comment:
I don't think tooltip works like that? The transformation isn't supposed to be done automatically? (ie, you have a real div with the tooltip content, and the js for tooltip does what is needed? I'm not sure, maybe ask Raphael/Vincent...)